### ISSUE TITLE
Don't focus the chart container when clicked through mouse

### ISSUE THREAD

**Issue by saurabhraj123** [recharts/recharts#4790]
## Do you want to request a feature or report a bug?
Feature

## What is the current behavior?
When accessibility layer is on and I click on the chart container, the container is focused. It should only focus on keyboard event.

Is it a valid request? If so, how can I that? I tried going through the codebase but unable to find exactly where this focus is being triggered  from. Help needed.

**Comment by ckifer** [comment]
I'm not sure I understand what you are describing. The container has to have focus in order to react to events. What about that is causing issues?

CC @julianna-langston

**Comment by saurabhraj123** [comment]
Showing the outlines when the chart is focused using keyboard event is fine..But when I mouse click on the chart, the outline is not ideal. 

<img width="1223" alt="Screenshot 2024-07-19 at 10 55 24 PM" src="https://github.com/user-attachments/assets/9ed10207-2af7-4bdb-bdc2-f9fc7bf26f90">

**Comment by ckifer** [comment]
This isn't our outline style, its the browser

Will wait for @julianna-langston to reply as shes a lot more knowledgeable about accessibility than I am 😅

**Comment by saurabhraj123** [comment]
The outline is provided by the browser on focused elements. I checked it by giving `outline: none` style to it and it did go away. My requirement is just to show the outline in case of keyboard events not for mouse events.

**Comment by julianna-langston** [comment]
If you style focusable elements with :focus, the outline will appear for mouse-based focus (clicking the element) or keyboard-based focus (tabbing to the element). 
What you want is probably :focus-visible, which will only display your focus styles for keyboard-based focus.

For more info, here’s a CSS Tricks article on the subject:
https://css-tricks.com/the-focus-visible-trick/

**Comment by saurabhraj123** [comment]
I get that.. But when I pass `accessibilityLayer: true` to the Charts, the default behaviour is to get focused on mouse events as well. How can I do that? Also, should we change re-charts default behaviour as well and focus it only for keyboard events?

**Comment by ckifer** [comment]
No I don't think so, it makes sense for the chart to be focused on click. As a non-screen reader user it's convenient to click on the chart and then be able to use arrow keys to navigate

**Comment by sKopheK** [comment]
> This isn't our outline or outline style, its the browser
> 
> Will wait for @julianna-langston to reply as shes a lot more knowledgeable about accessibility than I am 😅

did upgrade 2.1.9 ->2.12.7 a week ago and it started to appear, something must have been changed ^______^

**Comment by agoldis** [comment]
```css
.recharts-wrapper *:focus:not(:focus-visible) {
  outline: none !important;
}
```
